tools/release/publish.sh

```diff
@@ -80,17 +80,23 @@ CURRENT_DIR=$(dirname "${BASH_SOURCE[0]}")
 source "${CURRENT_DIR}"/version.sh
 CONFIG_TEMPLATE="${CURRENT_DIR}"/config_local_template.js
 
-echo "Filling latest=${DATALAB_VERSION}"
-sed -i -e s/{{DATALAB_VERSION_PLACEHOLDER}}/\"${DATALAB_VERSION}\"/ $CONFIG_TEMPLATE
-echo "Filling latest patch=${DATALAB_VERSION_PATCH}"
-sed -i -e s/{{DATALAB_VERSION_PATCH_PLACEHOLDER}}/\"${DATALAB_VERSION_PATCH}\"/ $CONFIG_TEMPLATE
-echo "Filling previous=${CURRENT_VERSION}"
-sed -i -e s/{{PREV_SEMVER_PLACEHOLDER}}/\"${CURRENT_VERSION}\"/ $CONFIG_TEMPLATE
-echo "Filling gtm account=${GTM_ACCOUNT}"
-sed -i -e s/{{GTM_ACCOUNT_PLACEHOLDER}}/\"${GTM_ACCOUNT}\"/ $CONFIG_TEMPLATE
+# Only if the current version will be updated, upload the new config_local.js file. We do
+# this for cases where two releases are published on the same day. If the new release's
+# PREVIOUS_SEMVER points to the older release with the same date, this results in a
+# loop for the rollback process.
+if [ "$CURRENT_VERSION" != "$DATALAB_VERSION" ]; then
+  echo "Filling latest=${DATALAB_VERSION}"
+  sed -i -e s/{{DATALAB_VERSION_PLACEHOLDER}}/\"${DATALAB_VERSION}\"/ $CONFIG_TEMPLATE
+  echo "Filling latest patch=${DATALAB_VERSION_PATCH}"
+  sed -i -e s/{{DATALAB_VERSION_PATCH_PLACEHOLDER}}/\"${DATALAB_VERSION_PATCH}\"/ $CONFIG_TEMPLATE
+  echo "Filling previous=${CURRENT_VERSION}"
+  sed -i -e s/{{PREV_SEMVER_PLACEHOLDER}}/\"${CURRENT_VERSION}\"/ $CONFIG_TEMPLATE
+  echo "Filling gtm account=${GTM_ACCOUNT}"
+  sed -i -e s/{{GTM_ACCOUNT_PLACEHOLDER}}/\"${GTM_ACCOUNT}\"/ $CONFIG_TEMPLATE
 
-gsutil cp ./config_local.js gs://${PROJECT_ID}/deploy/config_local_${BUILD}.js
-gsutil cp $CONFIG_TEMPLATE gs://${PROJECT_ID}/deploy/config_local.js
+  gsutil cp ./config_local.js gs://${PROJECT_ID}/deploy/config_local_${BUILD}.js
```

Comment:
./config_local.js is the current file which is about to become old, it's copied with the BUILD suffix. Copying the template is the step below.